lib/paths/index.ts

```diff
@@ -69,9 +77,9 @@ type RecursivePaths<
                             depth: Append<CallOptions["depth"]>;
                           }
                         > extends infer Rest
-                        ? IsNever<Rest> extends true
-                          ? never
-                          : `${Key}.${Rest & string}`
+                        ? Rest extends Exclude<Primitive, symbol>
```

Comment:
You may look for `Pathable` here:

```suggestion
                        ? Rest extends Pathable
```

I've tested it in https://github.com/ts-essentials/ts-essentials/pull/413 (commit https://github.com/ts-essentials/ts-essentials/pull/413/commits/98258d5fbccd4e6d0ba4d3e32ec58ce42c9f33ab) and it fixed the problem for TypeScript versions 4.5-4.8